libsquoosh/src/codecs.ts

```diff
@@ -9,6 +9,23 @@ import { cpus } from 'os';
   hardwareConcurrency: cpus().length,
 };
 
+interface DecodeModule extends EmscriptenWasm.Module {
+  decode: (data: Uint8Array) => any;
+}
+
+interface EncodeModule extends EmscriptenWasm.Module {
```

Comment:
I think types for encode modules for the codecs already exists: so maybe we can use them instead of using this EncodeModule (so that opts will be correctly typed for each codec).

Here is an example import for the MozJPEG codec:
```
import type {
  MozJPEGModule as MozJPEGEncodeModule,
  EncodeOptions as MozJPEGEncodeOptions,
} from '../../codecs/mozjpeg/enc/mozjpeg_enc';
```

Reply:
That seems like the correction solution but it won't work for me here:

![image](https://user-images.githubusercontent.com/229881/127586182-72623394-743a-4034-b781-bd2210f7aa9f.png)

Reply:
Hihiiim, it is strange. Here is what I tried and the only error we get is the return value says `Uint8Array | null`. So, we might need to handle `null` for the `encode` calls as well.

Here is my version of `decode.ts`: https://gist.github.com/ergunsh/93d0ddadd72d7f383fde952bbd45de31

WDYT?